test/common/ssl/ssl_socket_test.cc

```diff
@@ -2501,76 +2537,76 @@ TEST_P(SslSocketTest, ProtocolVersions) {
 
   // Connection using defaults (client & server) succeeds, negotiating TLSv1.2.
   testUtilV2(listener, client, "", true, "TLSv1.2", "", "", "", "", "ssl.handshake",
-             "ssl.handshake", GetParam());
+             "ssl.handshake", GetParam(), nullptr);
 
   // Connection using defaults (client & server) succeeds, negotiating TLSv1.2,
   // even with client renegotiation.
   client.set_allow_renegotiation(true);
   testUtilV2(listener, client, "", true, "TLSv1.2", "", "", "", "", "ssl.handshake",
-             "ssl.handshake", GetParam());
+             "ssl.handshake", GetParam(), nullptr);
   client.set_allow_renegotiation(false);
 
   // Connection using TLSv1.0 (client) and defaults (server) succeeds.
   client_params->set_tls_minimum_protocol_version(envoy::api::v2::auth::TlsParameters::TLSv1_0);
   client_params->set_tls_maximum_protocol_version(envoy::api::v2::auth::TlsParameters::TLSv1_0);
   testUtilV2(listener, client, "", true, "TLSv1", "", "", "", "", "ssl.handshake", "ssl.handshake",
-             GetParam());
+             GetParam(), nullptr);
 
   // Connection using TLSv1.1 (client) and defaults (server) succeeds.
   client_params->set_tls_minimum_protocol_version(envoy::api::v2::auth::TlsParameters::TLSv1_1);
   client_params->set_tls_maximum_protocol_version(envoy::api::v2::auth::TlsParameters::TLSv1_1);
   testUtilV2(listener, client, "", true, "TLSv1.1", "", "", "", "", "ssl.handshake",
-             "ssl.handshake", GetParam());
+             "ssl.handshake", GetParam(), nullptr);
 
   // Connection using TLSv1.2 (client) and defaults (server) succeeds.
   client_params->set_tls_minimum_protocol_version(envoy::api::v2::auth::TlsParameters::TLSv1_2);
   client_params->set_tls_maximum_protocol_version(envoy::api::v2::auth::TlsParameters::TLSv1_2);
   testUtilV2(listener, client, "", true, "TLSv1.2", "", "", "", "", "ssl.handshake",
-             "ssl.handshake", GetParam());
+             "ssl.handshake", GetParam(), nullptr);
```

Comment:
Maybe we should be using options objects to prevent this positional parameter growth? There's a lot of churn in this PR over just adding an additional parameter, which is kind of a sign that we have gone past the usefulness of positional parameters.

Reply:
Note that it is just a test helper function. Should we invest effort in its quality? Up to you to decide, I can refactor it as well while there.

Reply:
Up to you. It's a nice-to-have improvement in test maintainability, but if you just want to drop a TODO at the `testUtilv2` definition for now, that's fine.

Reply:
Let me put a TODO and fix it in a separate PR, for both `testUtil` and `testUtilV2`.

Reply:
Added TODO in https://github.com/envoyproxy/envoy/pull/4973/commits/2f231418932fea5b68f8cd36df8b781567b6fafa